x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -93,6 +95,12 @@ export interface BulkUpdateOptions<Params extends RuleTypeParams> {
   query: object | string | undefined | null;
 }
 
+export interface BulkUpdateCasesOptions {
```

Comment:
I think we could make an optimization here. The cases code will group the alerts by index such that we don't accidentally update an alert in the wrong index. And then we do a pmap resulting in multiple elasticsearch mget and bulk update requests.

Since we're already touching the alerts client do you think we could rely on the alerts client to accept id and index pairs? That way we'd only need a single mget to perform authorization and a single bulk update to do the changes.

To support that this interface could be something like:

```
{
  alerts: Array<{ ids: string[]; index: string; }>;
  caseIds: string[]
}
```

Or if we wanted to support different caseIds per group we'd basically pass in `BulkUpdateCasesOptions[]` (maybe change the name).